hack/make-and-check-size

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+#
+# make-and-check-size - wrapper around 'make' that also checks binary growth
+#
+# This script is intended to be run via 'git rebase -x', in a Makefile rule
+# such as:
+#
+#    build-all-new-commits:
+#       CONTEXT_DIR=$(shell mktemp -d --tmpdir make-size-check.XXXXXXX); \
+#       git rebase $(GIT_BASE_BRANCH)^ -x "hack/make-and-check-size $$CONTEXT_DIR"; \
+#       $(RM) -rf $$CONTEXT_DIR
+#
+# ...which has long been a part of our usual CI, one that makes sure that
+# each commit (in a multi-commit PR) can be compiled individually. By
+# adding the '^' to GIT_BASE_BRANCH we establish a baseline and store
+# the binary sizes of each file (podman, podman-remote) prior to our PR.
+#
+# CONTEXT_DIR is a temporary directory used to store the original sizes
+# of each binary file under bin/
+#
+# *IMPORTANT NOTE*: this script will leave the git checkout in a funky state!
+# (because we rebase onto a nonterminal commit). I believe this is OK, since
+# this makefile target is used only in CI and only in a scratch VM. Running
+# this in a development environment would yield unpredictable results anyway,
+# by rebasing onto origin/main by default and by leaving an aborted rebase
+# on failure.
+#
+ME=$(basename $0)
+
+# Maximum allowable size, in bytes. Can be overridden via github description
+MAX_BIN_GROWTH=$((100 * 1024))
```

Comment:
Review wanted: 
- is this a reasonable variable/override name?
- is this (100k) a reasonable value?

Reply:
Seems a bit low? Maybe 256k?

Reply:
Happy to change it but will wait til EOD in case there's discussion

Reply:
256k is to much IMO
Since I am the one currently going through this most unnecessary imports are already less than 100k.
With 256k 4 PRs could already add up to one 1MB

Reply:
I wondered if you were going to say that.  Lets take Paul's advice here.  It can always be changed.

Reply:
I guess the question here is how painful this should be.
There is a trade of here obviously. With a lower limit is will fail more often, likely on updating dependencies.
However I think a lower limit will prevent someone from import a new dependency which could bloat the binary over time. In this case we might want to look for a smaller alternative.

Reply:
I agree it should be small (even smaller than 100k), so we are aware of it.  If we still need the change we can always override it with `MAX_BIN_GROWTH=`

Reply:
FWIW, I like 100k to start.